### PR TITLE
feat(api): cover upload for books, read lists and collections

### PR DESCRIPTION
**Pull request by Snd-R**
Added the same functionality of custom series cover uploads to books, read lists and collections
no tests for now

### CODE REVIEW

File: komga/src/main/kotlin/org/gotson/komga/domain/persistence/ThumbnailSeriesCollectionRepository.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+interface ThumbnailSeriesCollectionRepository {
+  fun findByIdOrNull(thumbnailId: String): ThumbnailSeriesCollection?
+  fun findSelectedByCollectionIdOrNull(collectionId: String): ThumbnailSeriesCollection?
+  fun findAllByCollectionId(collectionId: String): Collection<ThumbnailSeriesCollection>
+
+  fun insert(thumbnail: ThumbnailSeriesCollection)
+  fun update(thumbnail: ThumbnailSeriesCollection)
+  fun markSelected(thumbnail: ThumbnailSeriesCollection)
+
+  fun delete(thumbnailBookId: String)
```
Comment (gotson) [inline komga/src/main/kotlin/org/gotson/komga/domain/persistence/ThumbnailSeriesCollectionRepository.kt:14]: should be `thumbnailCollection Id`

**Review by gotson — CHANGES_REQUESTED** [review]
Thanks for the contribution!

I see one thing missing, you need to delete the thumbnails before deleting the Collection or ReadList, else the delete would fail because of foreign keys. This is already done for Books and Series, so you can take inspiration there.

**Comment by gotson** [comment]
FYI some commits on `master` have added some Flyway migrations, you will need to adjust the filenames of your migrations to be after the existing ones.

**Review by gotson — CHANGES_REQUESTED** [review]
`BookLifecycle.addThumbnailForBook` is called by the REST controller, but also by `LocalArtworkLifecycle`. You need to handle the marking as selected differently for local assets, you can see what has been done for series, using `MarkSelectedPreference.IF_NONE_EXIST` [here](https://github.com/gotson/komga/blob/ef399a77ba7490783281dbf4cd6ca54eb6ea642a/komga/src/main/kotlin/org/gotson/komga/domain/service/SeriesLifecycle.kt#L267-L268).

**Comment by Snd-R** [comment]
With the current implementation when a new sidecar is detected it won't be marked as selected and user will have to somehow manually mark it
Should then new SIDECAR thumbnail be marked as selected when there's only one GENERATED thumbnail?

**Comment by gotson** [comment]
> With the current implementation when a new sidecar is detected it won't be marked as selected and user will have to somehow manually mark it Should then new SIDECAR thumbnail be marked as selected when there's only one GENERATED thumbnail?

You mean for books? Can we reproduce that behavior in a failing test?

We could probably change "if none exist" to something like "if none or default".

**Review by gotson — APPROVED** [review]

**Pull request merged**